lib/3scale/backend/listener.rb

```diff
@@ -457,11 +451,12 @@ def valid_usage_params?
         params[:usage].nil? || params[:usage].is_a?(Hash)
       end
 
-      def check_params_value_encoding!(input_params, params_to_validate)
-        params_to_validate.each do |p|
-          param_value = input_params[p]
-          if !param_value.nil? && !param_value.valid_encoding?
-            halt 400, ThreeScale::Backend::NotValidData.new.to_xml
+      def check_params_encoding!(input_params)
+        input_params.each do |key, value|
+          raise ArgumentError, Rack::ExceptionCatcher::INVALID_BYTE_SEQUENCE_ERR_MSG unless key.valid_encoding?
+
+          if !value.nil? && !value.valid_encoding?
```

Comment:
A little bit easier to read.
```suggestion
          if value && !value.valid_encoding?
```

But just a suggestion. Feel free to merge as you wish.

Reply:
This would behave differently when `value` is `false`, check: https://stackoverflow.com/questions/18060790/ruby-if-object-nil-or-if-object.